lexxpluss_apps/src/uss_controller.cpp

```diff
@@ -84,51 +84,58 @@ class uss_fetcher {
             if (!device_is_ready(dev[1]))
                 return -1;
         }
+
+        k_mutex_init(&lock);
         return 0;
     }
     void get_distance(uint32_t (&distance)[2]) const {
+        k_mutex_lock(&this->lock, K_FOREVER);
         distance[0] = this->distance[0];
         distance[1] = this->distance[1];
+        k_mutex_unlock(&this->lock);
     }
-    static void runner(void *p1, void *p2, void *p3) {
-        uss_fetcher *self{static_cast<uss_fetcher*>(p1)};
-        self->run();
-    }
-    k_thread thread;
-private:
-    void run() {
+    void update_once() {
         if (!device_is_ready(dev[0]))
             return;
-        while (true) {
-            if (sensor_sample_fetch_chan(dev[0], SENSOR_CHAN_ALL) == 0) {
+
+        if (sensor_sample_fetch_chan(dev[0], SENSOR_CHAN_ALL) == 0) {
+            sensor_value v;
+            sensor_channel_get(dev[0], SENSOR_CHAN_DISTANCE, &v);
+            int32_t value{v.val1 * 1000 + v.val2 / 1000};
+
+            k_mutex_lock(&this->lock, K_FOREVER);
+            distance[0] = distance[0] / 4 + value * 3 / 4;
```

Comment:
@sdswoops824 cc: @ar90n 
Maybe this is filtering? This is not relevant to this PR but it might be better to integrate this procedure to LexxAuto side.